src/components/negotiations-panel.tsx

```diff
@@ -0,0 +1,256 @@
+import TreeItem from '@material-ui/lab/TreeItem';
+import { SpotlightValue } from './spotlight-value';
+import { 
+  NegotiationConflict,
+  NegotiationStatus,
+  ResolveState
+} from '../negotiation-status-manager';
+
+import {
+  RobotTrajectoryManager,
+  NegotiationTrajectoryResponse
+} from '../robot-trajectory-manager';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    height: 240,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+  finished: {
+    backgroundColor: "lightgreen"
+  },
+  rejected: {
+    backgroundColor: theme.palette.warning.main
+  },
+  forfeited: {
+    backgroundColor: theme.palette.warning.main
+  },
+  defunct: {
+    backgroundColor: theme.palette.error.main
+  },
+  unresolved: {
+    backgroundColor: theme.palette.error.main
+  },
+  ongoing: {
+    backgroundColor: "yellow"
+  }
+}));
+
+interface Parameter
+{
+  conflict_version : number;
+  sequence : number[];
+};
+
+export interface NegotiationsPanelProps {
+  conflicts : Readonly<Record<string, NegotiationConflict>>;
+  spotlight?: Readonly<SpotlightValue<string>>;
+  trajManager?: Readonly<RobotTrajectoryManager>;
+  negotiationTrajStore : Record<string, NegotiationTrajectoryResponse>;
+}
+
+export default function NegotiationsPanel(props: NegotiationsPanelProps): JSX.Element {
+  const { conflicts, spotlight, trajManager, negotiationTrajStore } = props;
+
+  React.useEffect(() => {
+    if (!spotlight) {
+      return;
+    }
+    // TODO: spotlight
+  }, [spotlight]);
+  
+  const classes = useStyles();
+
+  /**  Utility conversion functions **/
+  const determineStatusText = (status : NegotiationStatus,
+    parent_resolved : ResolveState): string => {
+    if (status.forfeited)
+      return "  [FORFEITED]";
+    else if (status.rejected)
+      return "  [REJECTED]";
+    else if (status.defunct)
+      return "  [DEFUNCT]";
+    else if (parent_resolved === ResolveState.RESOLVED)
+      return "  [FINISHED]";
+    else
+      return "  [ONGOING]";
+  };
+  const determineStyle = (status : NegotiationStatus, 
+    parent_resolved : ResolveState): string => {
+    if (status.forfeited)
+      return classes.forfeited;
+    else if (status.rejected)
+      return classes.rejected;
+    else if (status.defunct)
+      return classes.defunct;
+    else if (parent_resolved === ResolveState.RESOLVED)
+      return classes.finished;
+    else
+      return classes.ongoing;
+  };
+
+  // keep track of parameters so we can send them as requests
+  let nodeidToParameters = new Map<string, Parameter>();
+
+  /** Render Negotiation panel contents **/
+  const renderNegotiations = (version : string, conflict : NegotiationConflict) => {
+    let conflictLabel = "Conflict #" + version + ", Participants: ";
+    let i = 0;
+    for (const name of Object.values(conflict.participantIdsToNames))
+    {
+      conflictLabel += name;
+
+      if (i !== (Object.keys(conflict.participantIdsToNames).length - 1))
+        conflictLabel += ", ";
+      ++i;
```

Comment:
What do you think about creating a function `getConflictLabel` ot handle this logic

```
 let conflictLabel = 'Conflict #' + version + ', Participants: ';
    let i = 0;
    for (const name of Object.values(conflict.participantIdsToNames)) {
      conflictLabel += name;

      if (i !== Object.keys(conflict.participantIdsToNames).length - 1) conflictLabel += ', ';
      ++i;
    }

```
 as you did with determineStyle?

Reply:
see https://github.com/osrf/romi-dashboard/pull/80#discussion_r486030501